doc/migration/version-upgrade-guide.md

```diff
@@ -168,8 +168,139 @@ The following sections outline the upgrade steps for each SDK version along with
     version = "=v0.7.1"
   ```
 
+## `v0.8.x`
+
+The SDK version `v0.8.x` supports scaffolding projects to use Go modules by default. It is recommended that you migrate your operator project to use modules for dependency management, however you can choose to keep using `dep`. The upgrade steps for both are outlined below:
+
+### `dep`
 
+- Update the SDK constraint in `Gopkg.toml` to version `v0.8.2`.
+  ```TOML
+  [[constraint]]
+    name = "github.com/operator-framework/operator-sdk"
+    version = "=v0.8.2"
+  ```
+- Pin the controller-tools dependency to the following revision. See the release notes or [#1278](https://github.com/operator-framework/operator-sdk/pull/1278/) for why this is needed.
+  ```TOML
+  [[override]]
+    name = "sigs.k8s.io/controller-tools"
+    revision = "9d55346c2bde73fb3326ac22eac2e5210a730207"
+  ```
+- Run `dep ensure` to update the vendor directory.
+
+### `modules`
+
+To get familiar with Go modules read the [modules wiki][modules-wiki]. In particular the section on [migrating to modules][migrating-to-modules].
+
+- Ensure that you have Go 1.12+ and [Mercurial][mercurial] 3.9+ installed.
+- Activate Go modules support for your project in `$GOPATH/src` by setting the env `GO111MODULES=on`. See [activating modules][activating-modules] for more details.
+- Initialize a new `go.mod` file by running `go mod init`.
+- Append the following to the end of your `go.mod` file to pin the operator-sdk and other upstream dependencies to the required versions.
+```
+// Pinned to kubernetes-1.13.1
+replace (
+	k8s.io/api => k8s.io/api v0.0.0-20181213150558-05914d821849
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20181213153335-0fe22c71c476
+	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20181127025237-2b1284ed4c93
+	k8s.io/client-go => k8s.io/client-go v0.0.0-20181213151034-8d9ed539ba31
+)
+
+replace (
+	github.com/coreos/prometheus-operator => github.com/coreos/prometheus-operator v0.29.0
+	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20181117043124-c2090bec4d9b
+	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20180711000925-0cf8f7e6ed1d
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.1.10
+	sigs.k8s.io/controller-tools => sigs.k8s.io/controller-tools v0.1.11-0.20190411181648-9d55346c2bde
+)
+
+replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.8.2
+```
```

Comment:
I think here we can add to remove the Go dep files.

Reply:
Can you outline why exactly?

Right now that's the last step once they've confirmed they can build the project with modules enabled.

Plus I think running `go mod tidy` shouldn't be affected by the presence of Gopkg.toml or the vendor directory.

Reply:
You are correct after I recheck it . Please ignore this comment.